code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -378,6 +379,41 @@
 	anchored = FALSE
 	target = null
 
+
+/mob/living/simple_animal/bot/cleanbot/larry/attackby(obj/item/I, mob/living/user)
+	if(istype(I, /obj/item/kitchen/knife) && !knife) //Is it a knife?
+		var/obj/item/kitchen/knife/newknife = I
+		knife = newknife
+		newknife.forceMove(src)
+		message_admins("[user] attached a [newknife.name] to [src]") //This should definitely be a notified thing.
+		AddComponent(/datum/component/larryknife, knife.force)
+		update_icons()
+	else
+		return ..()
+
+/mob/living/simple_animal/bot/cleanbot/larry/update_icons()
+	if(knife)
+		var/mutable_appearance/knife_overlay = knife.build_worn_icon(state = knife.icon_state, default_layer = 20, default_icon_file = 'icons/mob/inhands/misc/larry.dmi')
```

Comment:
This syntax isn't valid now.
`/obj/item/proc/build_worn_icon(default_layer = 0, default_icon_file = null, isinhands = FALSE, femaleuniform = NO_FEMALE_UNIFORM, override_state = null)`

Reply:
:tragedy:

Reply:
Wait where is the icon state determined?

Reply:
Webedit incoming